src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -214,6 +214,8 @@ def __init__(self, value: Any, datatype: Optional[DataType] = None) -> None:
             self.datatype = datatype
         else:
             self.datatype = infer_type(value)
+            if isinstance(self.datatype, StringType):
```

Comment:
@sfc-gh-aalam Are you good with this change?

Reply:
yes. approved the other PR which adds this change